xapian-letor/tests/api_letor.cc

```diff
@@ -33,32 +31,28 @@
 
 using namespace std;
 
-DEFINE_TESTCASE(assign_db_path1, !backend) {
-   Xapian::Ranker *ranker = new Xapian::ListNETRanker();
-   ranker->set_database_path("db_path");
-   TEST_EQUAL("db_path", ranker->get_database_path());
-   delete ranker;
-   ranker = new Xapian::SVMRanker();
-   ranker->set_database_path("db_path");
-   TEST_EQUAL("db_path", ranker->get_database_path());
-   delete ranker;
+DEFINE_TESTCASE(check_db_path, !backend) {
+   Xapian::ListNETRanker ranker1;
+   ranker1.set_database_path("db_path");
+   TEST_EQUAL("db_path", ranker1.get_database_path());
+   Xapian::SVMRanker ranker2;
+   ranker2.set_database_path("db_path");
+   TEST_EQUAL("db_path", ranker2.get_database_path());
 
    return true;
 }
 
-DEFINE_TESTCASE(assign_db_path2, backend) {
-    Xapian::Ranker *ranker = new Xapian::ListNETRanker();
-    TEST_EXCEPTION(Xapian::FileNotFoundError, ranker->train_model("", ""));
-    delete ranker;
-    ranker = new Xapian::SVMRanker();
-    TEST_EXCEPTION(Xapian::FileNotFoundError, ranker->train_model("", ""));
-    delete ranker;
+DEFINE_TESTCASE(check_train_model, backend) {
+    Xapian::ListNETRanker ranker1;
+    TEST_EXCEPTION(Xapian::FileNotFoundError, ranker1.train_model("", ""));
+    Xapian::SVMRanker ranker2;
+    TEST_EXCEPTION(Xapian::FileNotFoundError, ranker2.train_model("", ""));
 
     return true;
 }
 
 static void
-gen_uniqterms_gt_doclen_db(Xapian::WritableDatabase& db, const string&)
+populate_db(Xapian::WritableDatabase& db, const string&)
```

Comment:
While this no longer inaccurately describes the generated database, it now doesn't describe it at all.  When you need more generate functions, you'll end up calling them `populate_db2()` etc which really isn't helpful.

What are the particular feature(s) of this database?